.github/workflows/main.yml

```diff
@@ -0,0 +1,60 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-18.04
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        imgtag:
+          - "python:2.7-buster"
+          - "python:3.4-jessie"
+          - "python:3.5-buster"
+          - "python:3.6-buster"
+          - "python:3.7-buster"
+          - "python:3.8-buster"
+        tornado_ver:
+          - "4.5.3"
+          - "5.1.1"
+
+    container: "${{matrix.imgtag}}"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download NSQ
+        run: |
+          PY_VER="$(python -c 'import sys; print(sys.version_info[:2])')"
+          case "$PY_VER" in
+              '(3, 4)' | '(3, 7)' ) NSQ_DIST="nsq-1.0.0-compat.linux-amd64.go1.8" ;;
+              '(2, 7)' | '(3, 6)' ) NSQ_DIST="nsq-1.1.0.linux-amd64.go1.10.3" ;;
+              '(3, 5)' | '(3, 8)' ) NSQ_DIST="nsq-1.2.0.linux-amd64.go1.12.9" ;;
```

Comment:
notice my weird scheme to reduce the text matrix by correlating version of python and version of nsqd

Reply:
I did notice that, I resisted commenting 😁